feature/autodiscovery/api/src/main/kotlin/app/k9mail/autodiscovery/api/ConnectionSecurity.kt

```diff
@@ -0,0 +1,9 @@
+package app.k9mail.autodiscovery.api
+
+/**
+ * The connection security methods supported when using the [AutoDiscovery] mechanism.
+ */
+enum class ConnectionSecurity {
+    StartTls,
```

Comment:
Is the change from `uppercase underscore-separated naming` to `upper camel case` a general rule for enums? I stumbled over it as for `AuthenticationType` it is better readable and here it looks strange. It took me a bit to match `Tls` with `TLS`.

Reply:
The official Kotlin code style only states both options are valid. I find upper camel case generally easier to read.

This is an interesting case because `TLS` reads nicer than `Tls`. However, I prefer `StartTls` over `STARTTLS` or `START_TLS`.

Maybe in this case we make an exception and go with `TLS` and `StartTLS`?

Reply:
That's fine with me